syntax = "proto3";

package algotrader.model2;

enum DataType {
    DTFloat = 0;
    DTDouble = 1;
    DTInt32 = 2;
    DTInt64 = 3;
    DTBool = 4;
    DTString = 5;
    DTByteArray = 6;
}

message Series {


    string series_id = 1;
    string df_id = 2;
    string col_id = 3;
    string inst_id = 4;
    string provider_id = 5;
    DataType dtype = 6;

    // int64 start_time = 5;
    // int64 end_time = 6;
    //int64 length = 7;

    repeated int64 index = 11;

    repeated float float_data = 20;
    repeated double double_data = 21;
    repeated int32 int32_data = 22;
    repeated int64 int64_data = 23;
    repeated bool bool_data = 24;
    repeated string string_data = 25;
    repeated bytes bytes_data = 26;

}


message TimeSeriesItem {
    int64 timestamp = 1;
    map<string, double> data = 2;
}

message TimeSeriesUpdateEvent {
    string source = 1;
    TimeSeriesItem item = 2;
}
